### PR TITLE
Fixed some problem where test won't compile or fail

### PR DESCRIPTION
**Pull request by roy-powerfinance**
Fixed some test compile problem and failed tests

### CODE REVIEW

File: frame/support/src/dispatch.rs [hunk 1]
```diff
@@ -2075,7 +2075,7 @@ mod tests {
 		use super::*;
 
 		pub trait Trait {
-			type AccountId;
+			type AccountId: From<u32>;
```
Comment (aliXsed) [inline frame/support/src/dispatch.rs:2078]: With this change, you'd better remove the "where" constraint from:
`pub trait Trait: system::Trait + Sized where Self::AccountId: From<u32>`

File: frame/support/src/weights.rs [hunk 2]
```diff
@@ -335,7 +335,7 @@ mod tests {
 	mod system {
 		use super::*;
 		pub trait Trait {
-			type AccountId;
+			type AccountId: From<u32>;
```
Comment (aliXsed) [inline frame/support/src/weights.rs:338]: Same here.

**Comment by aliXsed** [comment]
@royyang-centrality regarding the update of dependencies in Cargo.lock, if you haven't intended them, I suggest reverting. I'm thinking perhaps we'd better update dependencies right after a release rather than close to a release.

File: frame/support/src/dispatch.rs (hunk 1 above)
Comment (roy-powerfinance) [reply frame/support/src/dispatch.rs:2078]: Yep will fix

**Review by aliXsed — APPROVED** [review]

**Pull request merged**